src/camera.rs

```diff
@@ -1,43 +1,89 @@
 //! Contains useful [`Camera`](struct.Camera.html) struct and `Projections`.
 use std::ops;
 
-use cgmath::{ortho as cgmath_ortho, perspective as cgmath_perspective, Deg};
+use cgmath;
 use mint;
 
 use NodePointer;
 use object::Object;
 
+/// A camera's projection.
+pub enum Projection {
+    /// An orthographic projection.
+    Orthographic(Orthographic),
+
+    /// A perspective projection.
+    Perspective(Perspective),
+}
+
 /// Camera is used to render Scene with specific `Projection`.
-pub struct Camera<P> {
+pub struct Camera {
     pub(crate) object: Object,
+
     /// Projection parameters of this camera.
-    pub projection: P,
+    pub projection: Projection,
 }
 
-impl<P> AsRef<NodePointer> for Camera<P> {
+impl Camera {
+    /// Computes the projection matrix representing the camera's projection.
+    pub fn project(&self, aspect_ratio: f32) -> mint::ColumnMatrix4<f32> {
+        self.projection.project(aspect_ratio)
+    }
+}
+
+impl Projection {
+    /// Constructs an orthographic projection.
+    pub fn orthographic<P: Into<mint::Point2<f32>>>(
+        center: P,
+        extent_y: f32,
+        near: f32,
+        far: f32,
+    ) -> Self {
+        Projection::Orthographic(Orthographic {
+            center: center.into(),
+            extent_y,
+            near,
+            far,
+        })
+    }
+
+    /// Constructs a perspective projection.
+    pub fn perspective(
+        fov_y: f32,
```

Comment:
I ran into a degrees vs. radians problem here. It might be worth altering the API to avoid in the future.

Reply:
Agreed, smth like `type Deg=f32` and `type Rad=f32` may be enough.